rasa_core_sdk/forms.py

```diff
@@ -443,7 +472,11 @@ def _validate_if_required(self, dispatcher, tracker, domain):
             - the form is called after `action_listen`
             - form validation was not cancelled
         """
-        if tracker.latest_action_name == "action_listen" and tracker.active_form.get(
+        if tracker.latest_action_name == "form" and tracker.active_form.get(
+            "validate", True
+        ):
```

Comment:
this if looks awful, is it black formatting?

Reply:
Did you test it? I doubt `tracker.latest_action_name == "form"` is correct

Reply:
I'll check again in a run to make sure, but it came from here:
```
    def _activate_if_required(self, tracker):
        # type: (Tracker) -> List[Dict]
        """Return `Form` event with the name of the form
            if the form was called for the first time"""

        if tracker.active_form.get("name") is not None:
            logger.debug("The form '{}' is active".format(tracker.active_form))
        else:
            logger.debug("There is no active form")

        if tracker.active_form.get("name") == self.name():
            return []
        else:
            logger.debug("Activated the form '{}'".format(self.name()))
            return [Form(self.name())]
```
and here:
```
class Form(Event):
    """If `name` is not None: activates a form with `name`
        else deactivates active form
    """
    type_name = "form"
```

Reply:
but `Form` is not `ActionExecuted` so it cannot be `latest_action_name`, also it is not part of the tracker yet, because it is returned by the very same call to `FormAction`

Reply:
You're totally right. Sorry about that. Here's what I suggest instead:
[This function](https://github.com/RasaHQ/rasa_core_sdk/blob/master/rasa_core_sdk/forms.py#L418) will only return anything but `[]` if the form becomes activated, correct? Then I would do:
```python3
        # activate the form
        events = self._activate_if_required(tracker)
        # validate pre-filled slots
        if events != [] and tracker.active_form.get("validate", True):
            events.extend(self._validate_prefilled_slots(dispatcher, tracker, domain))
```
I've checked and this does actually go into the `validate_prefilled_slots()` as expected. My only concern is that that activation function looks quite weird, i would expect it to `return []` after the logger statements in the first logic chunk too. Does this make sense to you?

Reply:
that's should be fine, I think, the `else` is triggered when the form is activated. I don't really like checking for empty event list, because we (or actually anyone in subclass) might change the return statement